ragna/core/_document.py

```diff
@@ -200,6 +200,14 @@ def supported_suffixes(cls) -> list[str]:
         """
         pass
 
+    @abc.abstractmethod
+    def mime_type(cls) -> str:
+        """
+        Returns:
+            MIME type of the document handler.
+        """
+        pass
```

Comment:
1. Can we have a reasonable default here instead of making it abstract? E.g. downloading the file instead of trying to display it.
2. Why is this on the handler on not on the document itself?

Reply:
Going to answer these out of order:
2) No reason at all, except that the document handler already parses file type from extension. Either the Document itself can also do that or can internally call DocumentHandler. Thoughts?
1) I don't think there's a default MIME type that will always trigger the browser to perform a download because the browser always needs some way to make sense of the incoming byte stream. If we put the method on Document though the default could be to try to parse the extension and raise a ValueError or RagnaException if it fails. I think I like that option.

Reply:
> No reason at all, except that the document handler already parses file type from extension. Either the Document itself can also do that or can internally call DocumentHandler. Thoughts?

Maybe we should make MIME-type just an optional parameter of `Document`. I like that more than putting it on the handler. Actually, the handler could also act on the MIME-type of the document rather than on the suffix. Of course that part does not need to happen in this PR. But IMO that makes more sense.

> I don't think there's a default MIME type that will always trigger the browser to perform a download

[RFC 2046 5.4.3](https://datatracker.ietf.org/doc/html/rfc2046#section-4.5.3)

> MIME implementations must at a minimum treat any unrecognized subtypes as being equivalent to "application/octet-stream".

That sounds like a reasonable default to me.

> If we put the method on Document though the default could be to try to parse the extension and raise a ValueError or RagnaException if it fails.

This ties into the first point of my comment. We could try to parse the MIME type from the suffix and if we can't determine it, we use `application/octet-stream`.